src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/core/java/codeaction/CodeActionHandler.java

```diff
@@ -195,15 +195,15 @@ public CodeAction resolveCodeAction(CodeAction unresolved, IPsiUtils utils) {
 
 		/*JavaCodeActionResolveContext context = new JavaCodeActionResolveContext(unit, start, end - start, utils, params,
 				unresolved);*/
-		JavaCodeActionResolveContext context = new JavaCodeActionResolveContext(unit.getViewProvider().clone().getPsi(unit.getLanguage()),
+		JavaCodeActionResolveContext context = new JavaCodeActionResolveContext(unit,
```

Comment:
@angelozerr can you please explain how these changes fix the issue at hand? Somehow the original code was cloning the view provider, get the psi from if while your new version would use the existing psi and copy the context (see further down). Correct? How does this fix the issue?